Minitwit_BE/Minitwit_BE.Api/Controllers/TwitController.cs

```diff
@@ -21,35 +21,60 @@ public async Task<string> TestEndpoint()
             return await Task.FromResult("test");
         }
 
-        // TODO: Adding to the database should be POST/PUT. The message should be taken from the request body.
-        [HttpGet("add")]
-        public async Task<string> AddTwit()
+        [HttpPost("add")]
+        public async Task<ActionResult<Message>> AddTwit([FromBody]MessageInput input)
         {
-            
             // Add new twit
             Console.WriteLine("Inserting a new twit");
             // Primary keys should be auto incremented when you add entity to the table and dont explicitely specify specify the ID
-            _twitContext.Add(new Message 
+            Message msg = new Message 
             { 
-                AuthorId = 1,
+                AuthorId = input.AuthorId,
                 Flagged = false,
                 PublishDate = DateTime.Now,
-                Text = "jjjjjjjjjjjjjjj"
-            });
-            _twitContext.SaveChanges();
+                Text = input.Text
+            };
             
+            _twitContext.Add(msg);
+            _twitContext.SaveChanges();
+            return Ok(msg);
+        }
 
-            return "OK";
+        [HttpGet("public-twits")]
+        public async Task<ActionResult<List<Message>>> GetTwits()
+        {
+            Console.WriteLine("Reading");
+            return Ok(_twitContext.Messages.ToList());
         }
 
-        [HttpGet("getall")]
-        public async Task<string> GetTwit()
+        [HttpGet("personal-twits/{id}")]
+        public async Task<ActionResult<List<Message>>> GetPersonalTwits(int id)
         {
-            // Print all to console
             Console.WriteLine("Reading");
-            _twitContext.Messages.OrderBy(m => m.MessageId).AsEnumerable().ToList().ForEach(e => Console.WriteLine($"Id: {e.MessageId}, Text: {e.Text}"));
+            return Ok(_twitContext.Messages.ToList().Where(msg => msg.AuthorId == id));
+        }
 
-            return "OK";
+        [HttpPut("mark-message")]
+        public async Task<ActionResult<string>> markMessage([FromBody]FlaggingInput input)
+        {
+            Message? flaggedMsg = _twitContext.Messages.SingleOrDefault(msg => msg.MessageId == input.MessageId);
+            if (flaggedMsg != null)
+            {
+                if (input.FlaggingAction == 1)
```

Comment:
I would change the interface for flagging. I would pass true if a person wants to flag a twit and false if doesn't want to flag the twit instead of 1 and 0. It's not really user friendly, what if for some reason it has a different value than 0 or 1?